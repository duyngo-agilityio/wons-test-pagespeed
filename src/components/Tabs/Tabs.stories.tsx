import type { Meta, StoryObj } from '@storybook/react';

// components
import Tabs from './index';

const meta: Meta<typeof Tabs> = {
  title: 'Components/Common/CustomTabs',
  component: Tabs,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
  argTypes: {
    tabs: {
      description: 'Array of tabs with dynamic key, label, and content',
    },
  },
};

export default meta;
type Story = StoryObj<typeof Tabs>;

export const Default: Story = {
  args: {
    tabs: [
      { key: 'event', label: 'Event', content: 'This is the event content' },
      {
        key: 'reminder',
        label: 'Reminder',
        content: 'This is the reminder content',
      },
      { key: 'task', label: 'Task', content: 'This is the task content' },
    ],
  },
};
