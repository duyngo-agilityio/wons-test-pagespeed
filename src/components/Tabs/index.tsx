'use client';

import { cn, Tabs as NextUITabs, Tab, TabsProps } from '@nextui-org/react';

interface TabType {
  key: string;
  label: string;
  content?: string;
  isDisable?: boolean;
}

interface CustomTabsProps extends TabsProps {
  tabs: TabType[];
}

const Tabs = ({ tabs, classNames, className, ...props }: CustomTabsProps) => {
  const { cursor, tabList, tab, tabContent } = classNames || {};

  return (
    <NextUITabs
      className={cn('p-4 rounded-[3px] dark:bg-purple-900', className)}
      aria-label="Custom tabs"
      classNames={{
        base: 'text-white',
        tabList: cn('bg-pink-400 dark:bg-pink-600', tabList),
        cursor: cn('bg-pink-500', cursor),
        tabContent: cn(
          'text-gray-700 dark:text-pink-700 group-data-[selected=true]:text-white',
          tabContent,
        ),
        tab: cn('my-[2px] mx-[2px]', tab),
      }}
      {...props}
    >
      {tabs.map((tab) => (
        <Tab
          key={tab.key}
          title={tab.label}
          isDisabled={tab.isDisable}
          className="px-4 py-2"
        />
      ))}
    </NextUITabs>
  );
};

export default Tabs;
