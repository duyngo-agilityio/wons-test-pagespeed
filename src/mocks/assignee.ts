import { TRole } from '@/types';

export const MOCK_ASSIGNEES = {
  data: [
    {
      id: 1,
      attributes: {
        name: 'John Doe',
        avatar: '/avatar1.jpg',
        email: 'john@example.com',
        username: 'john_doe',
        fullName: 'John Doe',
        password: 'password123',
        role: 'User' as unknown as TRole,
        token: 'token123',
      },
    },
    {
      id: 2,
      attributes: {
        name: 'Jane Smith',
        avatar: '/avatar2.jpg',
        email: 'jane@example.com',
        username: 'jane_smith',
        fullName: 'Jane Smith',
        password: 'password123',
        role: 'User' as unknown as TRole,
        token: 'token123',
      },
    },
  ],
};
