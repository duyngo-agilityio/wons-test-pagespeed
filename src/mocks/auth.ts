// Types
import { ISignUpFormData } from '@/types';

export const SIGN_UP_FORM_DATA_MOCK: ISignUpFormData = {
  fullName: 'John Doe',
  username: 'johndoe',
  email: 'john.doe@example.com',
  password: 'Abcd@0987',
};

export const MOCK_SUCCESS_SIGN_UP_RES = {
  data: { user: { ...SIGN_UP_FORM_DATA_MOCK, id: '1' } },
  totalCount: 0,
};
