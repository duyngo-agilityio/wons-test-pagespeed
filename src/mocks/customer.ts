// Models
import { ICustomer } from '@/models';

// Types
import { TableColumnType } from '@/types';

export const PRODUCT_TABLE_COLUMNS_MOCK: TableColumnType<Partial<ICustomer>>[] =
  [
    {
      header: 'Name',
      accessor: 'name',
      isSort: true,
    },
    {
      header: 'FirstName',
      accessor: 'firstName',
      isSort: true,
    },
    {
      header: 'LastName',
      accessor: 'lastName',
      isSort: true,
    },
    {
      header: 'Email',
      accessor: 'email',
      isSort: true,
    },
  ];

export const PRODUCT_MOCK = [
  {
    id: '1',
    name: 'John Doe',
    firstName: 'John',
    lastName: 'Doe',
    email: 'john.doe@example.com',
    phone: '123-456-7890',
    job: 'Software Engineer',
    address: '123 Main St, City, Country',
    avatar: 'https://example.com/avatar1.jpg',
  },
  {
    id: '2',
    name: 'Jane Smith',
    firstName: 'Jane',
    lastName: 'Smith',
    email: 'jane.smith@example.com',
    phone: '098-765-4321',
    job: 'Project Manager',
    address: '456 Elm St, City, Country',
    avatar: 'https://example.com/avatar2.jpg',
  },
  {
    id: '3',
    name: 'Alice Johnson',
    firstName: 'Alice',
    lastName: 'Johnson',
    email: 'alice.johnson@example.com',
    phone: '555-666-7777',
    job: 'UX Designer',
    address: '789 Oak St, City, Country',
    avatar: 'https://example.com/avatar3.jpg',
  },
  {
    id: '4',
    name: 'Bob Williams',
    firstName: 'Bob',
    lastName: 'Williams',
    email: 'bob.williams@example.com',
    phone: '111-222-3333',
    job: 'Data Scientist',
    address: '101 Pine St, City, Country',
    avatar: 'https://example.com/avatar4.jpg',
  },
  {
    id: '5',
    name: 'Charlie Brown',
    firstName: 'Charlie',
    lastName: 'Brown',
    email: 'charlie.brown@example.com',
    phone: '444-555-6666',
    job: 'Marketing Specialist',
    address: '202 Birch St, City, Country',
    avatar: 'https://example.com/avatar5.jpg',
  },
];
