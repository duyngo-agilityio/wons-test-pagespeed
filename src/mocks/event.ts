export const EVENT_MOCKS = {
  title: 'Test Event',
  status: 'active',
  location: 'New York',
  date: new Date('2024-10-14'),
  startTime: '10:00',
  endTime: '12:00',
  users_permissions_users: [1, 2],
};

export const EVENTS_MOCKS = [
  {
    id: 1,
    attributes: {
      title: 'Test Event',
      status: 'active',
      location: 'New York',
      date: new Date('2024-10-14'),
      startTime: '10:00',
      endTime: '12:00',
      users_permissions_users: [1, 2],
    },
  },

  {
    id: 2,
    attributes: {
      title: 'Implement Component',
      descriptions: 'Implement for all component',
      date: '2024-10-20T01:30:00Z',
      time: '2024-10-20T01:30:00Z',
      createdAt: '2024-10-22T09:46:11.546Z',
      updatedAt: '2024-10-22T09:46:16.769Z',
      publishedAt: '2024-10-22T09:46:16.768Z',
    },
  },
];

export const EVENTS_MOCKS_WIDTH_USERS_PERMISSIONS = {
  title: 'Test Event',
  status: 'active',
  location: 'New York',
  date: new Date('2024-10-14'),
  startTime: '10:00',
  endTime: '12:00',
  users_permissions_users: {
    data: [
      {
        id: 3,
        attributes: {
          username: 'superadmin',
          email: 'admin1@gmail.com',
          provider: 'local',
          confirmed: false,
          blocked: false,
          fullName: 'Super Admin',
          createdAt: '2024-08-28T04:29:52.000Z',
          updatedAt: '2024-10-11T02:14:38.208Z',
          avatar:
            'https://watermark.lovepik.com/photo/20211209/large/lovepik-japanese-fresh-girl-park-photo-picture_501698500.jpg',
        },
      },
    ],
  },
};
