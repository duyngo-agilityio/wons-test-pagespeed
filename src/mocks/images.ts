export const MOCK_IMAGES = [
  'https://media.wired.com/photos/6500ad57fe61eb702d721b58/1:1/w_1800,h_1800,c_limit/Apple-iPhone-15-Pro-Hero-Gear.jpg',
  'https://clickbuy.com.vn/uploads/images/2019/09/thumb_iP11_3.jpg',
];
