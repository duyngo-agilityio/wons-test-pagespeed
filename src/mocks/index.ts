export * from './customer';
export * from './tabs';
