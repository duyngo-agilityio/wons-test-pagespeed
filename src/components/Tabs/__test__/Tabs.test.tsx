// components
import Tabs from '../index';

// mocks
import { EVENT_TABS } from '@/mocks';

describe('CustomTabs component', () => {
  it('renders correctly and matches snapshot', () => {
    const { container } = testLibJestUtils.render(<Tabs tabs={EVENT_TABS} />);
    expect(container).toMatchSnapshot();
  });
});
