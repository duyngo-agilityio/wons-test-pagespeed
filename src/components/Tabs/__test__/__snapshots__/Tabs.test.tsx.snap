// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTabs component renders correctly and matches snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <div
      class="p-4 rounded-xl  dark:bg-purple-900"
    >
      <div
        aria-label="Custom tabs"
        class="inline-flex text-white"
        data-slot="base"
      >
        <div
          aria-label="Custom tabs"
          aria-orientation="horizontal"
          class="flex p-1 h-fit gap-2 items-center flex-nowrap overflow-x-scroll scrollbar-hide rounded-medium bg-[#fee9f3] dark:bg-[#714e6e]"
          data-slot="tabList"
          id="react-aria-:r0:"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="z-0 w-full flex group relative justify-center items-center cursor-pointer transition-opacity tap-highlight-transparent data-[disabled=true]:cursor-not-allowed data-[disabled=true]:opacity-30 data-[hover-unselected=true]:opacity-disabled outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 h-8 text-small rounded-small my-[2px] mx-[2px] px-4 py-2"
            data-key="event"
            data-selected="true"
            data-slot="tab"
            id="react-aria-:r0:-tab-event"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="absolute z-0 inset-0 rounded-small dark:bg-default shadow-small bg-[#FF69B4]"
              data-slot="cursor"
            />
            <div
              class="relative z-10 whitespace-nowrap transition-colors text-[#515567] dark:text-[#d4c9d3] group-data-[selected=true]:text-white"
              data-slot="tabContent"
            >
              Event
            </div>
          </button>
          <button
            aria-selected="false"
            class="z-0 w-full flex group relative justify-center items-center cursor-pointer transition-opacity tap-highlight-transparent data-[disabled=true]:cursor-not-allowed data-[disabled=true]:opacity-30 data-[hover-unselected=true]:opacity-disabled outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 h-8 text-small rounded-small my-[2px] mx-[2px] px-4 py-2"
            data-key="reminder"
            data-slot="tab"
            id="react-aria-:r0:-tab-reminder"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <div
              class="relative z-10 whitespace-nowrap transition-colors text-[#515567] dark:text-[#d4c9d3] group-data-[selected=true]:text-white"
              data-slot="tabContent"
            >
              Reminder
            </div>
          </button>
          <button
            aria-selected="false"
            class="z-0 w-full flex group relative justify-center items-center cursor-pointer transition-opacity tap-highlight-transparent data-[disabled=true]:cursor-not-allowed data-[disabled=true]:opacity-30 data-[hover-unselected=true]:opacity-disabled outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 h-8 text-small rounded-small my-[2px] mx-[2px] px-4 py-2"
            data-key="task"
            data-slot="tab"
            id="react-aria-:r0:-tab-task"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <div
              class="relative z-10 whitespace-nowrap transition-colors text-[#515567] dark:text-[#d4c9d3] group-data-[selected=true]:text-white"
              data-slot="tabContent"
            >
              Task
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
